frontend/src/Editor/Components/Table/Table.jsx

```diff
@@ -455,6 +455,21 @@ export function Table({
     },
     [serverSidePagination, clientSidePagination, setPaginationInternalPageIndex]
   );
+  registerAction(
+    'selectRow',
+    async function (key, value) {
+      const item = currentState.components[component.name]['currentData'].filter((item) => item[key] == value);
+      const row = rows.find((item, index) => item.original[key] == value);
+      if (row != undefined) {
+        const selectedRowDetails = { selectedRow: item[0], selectedRowId: row.id };
+        mergeToTableDetails(selectedRowDetails);
+        setExposedVariables(selectedRowDetails).then(() => {
+          fireEvent('onRowClicked');
+        });
+      }
+    },
+    []
```

Comment:
Try adding `mergeToTableDetails` as dependency here.

Otherwise, if you add a table inside a modal, and close the modal, and open it again, and then run the csa, the highlight won't work.

Reply:
Please also ensure that the above case doesn't fail.